packages/lesswrong/components/ultraFeed/UltraFeedItemFooter.tsx

```diff
@@ -0,0 +1,179 @@
+import { getVotingSystemByName } from "@/lib/voting/votingSystems";
+import { getNormalizedReactionsListFromVoteProps } from "@/lib/voting/namesAttachedReactions";
+
+const styles = defineStyles("UltraFeedItemFooter", (theme: ThemeType) => ({
+  root: {
+    position: "relative",
+    paddingLeft: 12,
+    paddingRight: 12,
+    paddingTop: 8,
+    paddingBottom: 8,
+    display: "flex",
+    flexWrap: "wrap",
+    alignItems: "center",
+    justifyContent: "space-between",
+    color: `${theme.palette.text.dim3} !important`,
+    opacity: `1 !important`,
+    fontFamily: theme.palette.fonts.sansSerifStack,
+    fontSize: "1.3rem !important",
+    "& *": {
+      color: `${theme.palette.text.dim3} !important`,
+    },
+    "& svg": {
+      color: `${theme.palette.text.dim3} !important`,
+    },
+    "& a:hover, & a:active": {
+      textDecoration: "none",
+      color: `${theme.palette.linkHover.dim} !important`,
+    },
+  },
+  commentCount: {
+    display: "flex",
+    alignItems: "center",
+    paddingBottom: 2,
+    "& svg": {
+      height: 22,
+    },
+  },
+  commentCountClickable: {
+    cursor: "pointer",
+    "&:hover": {
+      color: theme.palette.grey[1000],
+    },
+  },
+  commentCountText: {
+    marginLeft: 4,
+  },
+  addReactionButton: {
+    display: 'flex',
+    margin: '0 6px',
+    alignItems: 'center',
+    '& .react-hover-style': {
+      filter: 'opacity(1) !important',
+    },
+    '& svg': {
+      filter: 'opacity(1) !important',
+      height: 22,
+      width: 22,
+    }
+  },
+  reactionIcon: {
+    marginRight: 6,
+  },
+  reactionCount: {
+    marginTop: -2
+  },
+  bookmarkButton: {
+    marginBottom: -2,
+  }
+}));
+
+type DocumentType = PostsListWithVotes | CommentsList;
+
+interface UltraFeedItemFooterProps {
+  document: DocumentType;
+  collectionName: "Posts" | "Comments"; // Now required, no longer optional
+}
+
+const UltraFeedItemFooter = ({
+  document,
+  collectionName,
+}: UltraFeedItemFooterProps) => {
+  const classes = useStyles(styles);
+  const { BookmarkButton, OverallVoteAxis, AgreementVoteAxis, AddReactionButton } = Components;
+
+  const isComment = collectionName === "Comments";
+  
+  const parentPost = isComment ? (document as UltraFeedComment).post : undefined;
+  
+  const commentCount = isComment 
+    ? (document as CommentsList).descendentCount || 0 
+    : (document as PostsListWithVotes).commentCount || 0;
+    
+  const commentsAreClickable = true;
+
+  // TODO: Implement this
+  const onClickComments = () => {
+  };
+
+  const commentCountIcon = (
+    <div
+      onClick={onClickComments}
+      className={classNames(classes.commentCount, {
+        [classes.commentCountClickable]: commentsAreClickable,
+      })}
+    >
+      <CommentIcon />
+      <span className={classes.commentCountText}>
+        {commentCount}
+      </span>
+    </div>
+  );
+
+  const votingSystemName = isComment ? parentPost?.votingSystem : (document as PostsListWithVotes)?.votingSystem;
```

Comment:
The number of casts and amount of conditional logic based on whether this is a post or comment makes me think this should have been two components... or otherwise somehow structured quite differently, if a single component.

Reply:
now rewritten